### PR TITLE
CorelliPowderCalibrationCreate: Implement rotational constraint

### PR DESCRIPTION
**Pull request by jmborr**
**Description of work.**
- [x] Option in `AlignComponents` to output table listing changes in position and orientation for each bank
- [x] Update documentation in `AlignComponents`
- [x] Option in `CorelliPowderCalibrationCreate` to prevent rotations of the bank around its normal axis
- [x] Update documentation in `CorelliPowderCalibrationCreate`
- [x] Unit test for fixed roation
- [x] Preserve existing system test with original options
- [x] Updated release notes

**To test:**  
In the analysis machines, run this scripts (takes ~15 minutes)
```python
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np
from corelli.calibration import load_calibration_set, apply_calibration
from corelli.calibration.powder import load_and_rebin

##############################################
# CREATE A POWDER CALIBRATION WITH Si SAMPLE
##############################################
Load(Filename='CORELLI_165752-165763',  # Silicon powder runs
     OutputWorkspace='Si',
     BankName='bank52, bank70', # only for testing purposes
     LoadLogs=False)

##############################################
# LOAD AND APPLY THE TUBE CALIBRATION
##############################################
load_calibration_set(input_workspace='Si', database_path='/SNS/CORELLI/shared/calibration/tube', output_calibration_name='tube_calibration')
apply_calibration(workspace='Si', output_workspace='Si', calibration_table='tube_calibration')

# Workspace Si_uncalibrated contains just the summed intensities and the uncalibrated instrument. Saves memory
Rebin(InputWorkspace='Si', OutputWorkspace='Si_uncalibrated', Params=[3000,13600,16660], PreserveEvents=False)

##############################################
# RUN THE POWDER CALIBRATION
##############################################
si_dspacings = [0.9179, 0.9600, 1.0451, 1.2458, 1.3576, 1.5677, 1.6374, 3.1353]
CorelliPowderCalibrationCreate(InputWorkspace='Si',
                               ComponentList='bank52/sixteenpack,bank70/sixteenpack',
                               TofBinning=[3000,-0.001,16660],
                               PeakPositions=si_dspacings,
                               OutputWorkspacesPrefix='Si_',
                               FixSource=True,
                               SourceToSampleDistance=20.04,
                               ComponentMaxTranslation=0.02,  # 2 cm
                               FixY=True,  # no displacements of the banks along the vertical axis
                               ComponentMaxRotation=3,  # 3 degree
                               FixYaw=True, # no rotations around the axis normal to the bank
                               Minimizer='L-BFGS-B')
# Workspace Si contains just the summed intensities and the uncalibrated instrument. Saves memory
Rebin(InputWorkspace='Si', OutputWorkspace='Si', Params=[3000,13600,16660], PreserveEvents=False)  # calibrated instrument
```
In table `Si_displacements`, check that `DeltaGamma` is zero for both banks

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by KedoKudo — COMMENTED** [review]
The testing script runs and produces expected results on the analysis cluster. I would recommend consolidating the terminology used across the affected algorithms so that the user won have to guess the meaning of each term.

File: Framework/PythonInterface/plugins/algorithms/AlignComponents.py [hunk 1]
```diff
@@ -28,6 +28,7 @@ class AlignComponents(PythonAlgorithm):
     _optionsList = ["Xposition", "Yposition", "Zposition", "AlphaRotation", "BetaRotation", "GammaRotation"]
     adjustment_items = ['ComponentName', 'Xposition', 'Yposition', 'Zposition',
                         'XdirectionCosine', 'YdirectionCosine', 'ZdirectionCosine', 'RotationAngle']
+    displacement_items = ['ComponentName', 'DeltaR', 'DeltaX', 'DeltaY', 'DeltaZ', 'DeltaAlpha', 'DeltaBeta', 'DeltaGamma']
```
[the remark this answers is not in the document]
Comment (jmborr) [reply Framework/PythonInterface/plugins/algorithms/AlignComponents.py:31]: comments added

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/CorelliPowderCalibrationCreate.py [hunk 2]
```diff
@@ -145,9 +145,10 @@ def PyInit(self):
                              doc='Comma separated list on banks to refine')
         self.declareProperty(name='ComponentMaxTranslation', defaultValue=0.02,
                              doc='Maximum translation of each component along either of the X, Y, Z axes (m)')
+        self.declareProperty(name='FixYaw', defaultValue=True, doc="Prevent rotations around the axis normal to the bank")
```
[the remark this answers is not in the document]
Comment (jmborr) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/CorelliPowderCalibrationCreate.py:148]: The "Yaw" specifies rotations around the axis _normal_ to an airplane. That's why I chose the name :smile: 
https://en.wikipedia.org/wiki/Aircraft_principal_axes

**Review by KedoKudo — APPROVED** [review]
Looks good, happy to approve :+1:

**Comment by jmborr** [comment]
@KedoKudo @peterfpeterson thank you 😄

**Pull request merged**